### PR TITLE
fix: updates to error handler print format

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- prints the deepest amplify exception only
- on debug prints third party exception

#### Description of how you validated changes
- manual changes

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/domain/amplify-usageData/NoUsageData.ts [hunk 1]
```diff
@@ -53,6 +53,13 @@ export class NoUsageData implements IUsageData, IFlowData {
     /* noop */
   }
 
+  /**
+   * Noop implementation of getSessionUuid
+   */
+  getSessionUuid(): string {
+    return '';
+  }
```
Comment (awsluja) [inline packages/amplify-cli/src/domain/amplify-usageData/NoUsageData.ts:61]: todo?

Comment (lazpavel) [reply packages/amplify-cli/src/domain/amplify-usageData/NoUsageData.ts:61]: no, this is a mock UsageData, empty string is ok

**Review by awsluja — DISMISSED** [review]
looks good; can you run e2e?

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-cli/src/amplify-exception-handler.ts [hunk 2]
```diff
@@ -37,55 +37,55 @@ export const handleException = async (exception: unknown): Promise<void> => {
...
   process.exitCode = 1;
 };
 
+const getDeepestAmplifyException = (amplifyException: AmplifyException): AmplifyException => {
+  let deepestAmplifyException = amplifyException;
+  while (deepestAmplifyException.downstreamException && amplifyException.downstreamException instanceof AmplifyException) {
+    deepestAmplifyException = amplifyException.downstreamException;
+  }
+  return deepestAmplifyException;
+};
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply packages/amplify-cli/src/amplify-exception-handler.ts:87]: changed to iteration

File: packages/amplify-cli/src/amplify-exception-handler.ts [hunk 3]
```diff
@@ -37,55 +37,55 @@ export const handleException = async (exception: unknown): Promise<void> => {
...
     }),
     'Failed to log error',
   );
 
   process.exitCode = 1;
 };
 
+const getDeepestAmplifyException = (amplifyException: AmplifyException): AmplifyException => {
+  let deepestAmplifyException = amplifyException;
+  while (deepestAmplifyException.downstreamException && amplifyException.downstreamException instanceof AmplifyException) {
```
Comment (jhockett) [inline packages/amplify-cli/src/amplify-exception-handler.ts:83]: Doesn't the condition need to check `deepestAmplifyException.downstreamException`?

```suggestion
  while (deepestAmplifyException.downstreamException && deepestAmplifyException.downstreamException instanceof AmplifyException) {
```

Comment (lazpavel) [reply packages/amplify-cli/src/amplify-exception-handler.ts:83]: good catch

**Review by jhockett — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**